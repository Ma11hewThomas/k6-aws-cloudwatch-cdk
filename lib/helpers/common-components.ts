import { GraphWidget, Metric } from '@aws-cdk/aws-cloudwatch';
import { Duration } from '@aws-cdk/core';

export function k6Widget(
  title: string,
  leftMetrics: Array<Metric>,
  stacked?: boolean,
): GraphWidget {
  return new GraphWidget({
    title: title,
    width: 24,
    height: 6,
    stacked: stacked ?? true,
    liveData: true,
    left: leftMetrics,
  });
}

export function k6Metric(
  name: string,
  label: string,
  type = 'timing',
  statistic = 'avg',
): Metric {
  return new Metric({
    namespace: 'k6',
    metricName: name,
    label: label,
    // @ts-ignore
    period: Duration.seconds(1),
    dimensions: {
      metric_type: type,
    },
    statistic: statistic,
  });
}
