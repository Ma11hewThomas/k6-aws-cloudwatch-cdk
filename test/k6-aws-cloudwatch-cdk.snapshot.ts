import { SynthUtils } from '@aws-cdk/assert';
import { createStack } from './helpers/helpers';

test('match snapshot', () => {
  expect(SynthUtils.toCloudFormation(createStack())).toMatchSnapshot();
});
