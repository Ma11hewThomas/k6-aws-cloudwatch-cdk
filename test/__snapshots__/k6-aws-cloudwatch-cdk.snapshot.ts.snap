// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match snapshot 1`] = `
Object {
  "Resources": Object {
    "DashCCD7F836": Object {
      "Properties": Object {
        "DashboardBody": Object {
          "Fn::Join": Array [
            "",
            Array [
              "{\\"widgets\\":[{\\"type\\":\\"text\\",\\"width\\":24,\\"height\\":1,\\"x\\":0,\\"y\\":0,\\"properties\\":{\\"markdown\\":\\"# K6\\"}},{\\"type\\":\\"metric\\",\\"width\\":24,\\"height\\":6,\\"x\\":0,\\"y\\":1,\\"properties\\":{\\"view\\":\\"timeSeries\\",\\"title\\":\\"Virtual users\\",\\"region\\":\\"",
              Object {
                "Ref": "AWS::Region",
              },
              "\\",\\"stacked\\":true,\\"metrics\\":[[\\"k6\\",\\"k6_vus\\",\\"metric_type\\",\\"gauge\\",{\\"label\\":\\"Virtual users\\",\\"period\\":1}]],\\"yAxis\\":{},\\"liveData\\":true}},{\\"type\\":\\"metric\\",\\"width\\":24,\\"height\\":6,\\"x\\":0,\\"y\\":7,\\"properties\\":{\\"view\\":\\"timeSeries\\",\\"title\\":\\"Average response time\\",\\"region\\":\\"",
              Object {
                "Ref": "AWS::Region",
              },
              "\\",\\"stacked\\":true,\\"metrics\\":[[\\"k6\\",\\"k6_http_req_sending\\",\\"metric_type\\",\\"timing\\",{\\"label\\":\\"Sending\\",\\"period\\":1}],[\\"k6\\",\\"k6_http_req_receiving\\",\\"metric_type\\",\\"timing\\",{\\"label\\":\\"Receiving\\",\\"period\\":1}],[\\"k6\\",\\"k6_http_req_handshaking\\",\\"metric_type\\",\\"timing\\",{\\"label\\":\\"Handshaking\\",\\"period\\":1}],[\\"k6\\",\\"k6_http_req_connecting\\",\\"metric_type\\",\\"timing\\",{\\"label\\":\\"Connecting\\",\\"period\\":1}],[\\"k6\\",\\"k6_http_req_waiting\\",\\"metric_type\\",\\"timing\\",{\\"label\\":\\"Waiting\\",\\"period\\":1}],[\\"k6\\",\\"k6_http_req_blocked\\",\\"metric_type\\",\\"timing\\",{\\"label\\":\\"Blocked\\",\\"period\\":1}]],\\"yAxis\\":{},\\"liveData\\":true}}]}",
            ],
          ],
        },
        "DashboardName": "k6-aws-cloudwatch-cdk",
      },
      "Type": "AWS::CloudWatch::Dashboard",
    },
  },
}
`;
